lib/new_relic/agent/local_log_decorator.rb

```diff
@@ -0,0 +1,28 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic
+  module Agent
+    #
+    # This module contains helper methods related to decorating log messages
+    module LocalLogDecorator
+      extend self
+
+      def decorate(message)
+        return message unless decorating_enabled?
+        metadata = NewRelic::Agent.linking_metadata
+        "#{message} NR-LINKING|#{metadata["entity.guid"]}|#{metadata["hostname"]}|#{metadata["trace.id"]}|#{metadata["span.id"]}|"
+      end
+
+      private
+
+      def decorating_enabled?
+        # :'application_logging.enabled' &&
```

Comment:
PR #1007 has not been merged yet, so the setting doesn't exist. I plan to add this back once that PR is merged, unless anyone has objections!